jedi/inference/compiled/subprocess/__init__.py

```diff
@@ -181,14 +181,20 @@ def _get_process(self):
             os.path.dirname(os.path.dirname(parso_path)),
             '.'.join(str(x) for x in sys.version_info[:3]),
         )
+        # Use explicit envionment to ensure reliable results (#1540)
+        env = {}
+        if os.name == 'nt':
+            # Windows requires SYSTEMROOT
+            env.update(SYSTEMROOT=os.environ.get('SYSTEMROOT'))
```

Comment:
Minor issue: While SYSTEMROOT is probably defined most of the time, if it is not we should just not set it. The problem is essentially that it could be empty. If it is we cannot set it:
```
>>> os.environ['asdf'] = None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.5/os.py", line 730, in __setitem__
    value = self.encodevalue(value)
  File "/usr/lib/python3.5/os.py", line 798, in encode
    raise TypeError("str expected, not %s" % type(value).__name__)
TypeError: str expected, not NoneType
```

Reply:
Another issue: Windows environment variables are case insensitive, so it also be `SYSTEMROOT`, `SystemRoot`, or maybe `systemroot`. I'll put in some robustness and if none are present, I'll refrain from setting.